Function of npm metadata merging into S3

yma88 opened the pull request.

hjmodi commented:
> This pull request **introduces 2 alerts** when merging [6c824fe](https://github.com/Commonjava/mrrc-uploader/commit/6c824feb8ed0f3c28851a3d7907b02ae6831e5cc) into [52a797b](https://github.com/Commonjava/mrrc-uploader/commit/52a797bb333dc38293a41aae3d5c358ab5508bdc) - [view on LGTM.com](https://lgtm.com/projects/g/Commonjava/mrrc-uploader/rev/pr-698fa31fb8c0e23e11e11d1f4c5542a3dafafbd7)
> 
> **new alerts:**
> 
> * 2 for Module is imported with 'import' and 'import from'

Hi @yma88 , I've added this LGTM analysis to find any issues in the code automatically. Currently it shows two issues please take a look at them :)

yma88 commented:
@hjmodi This is cool! Thank you, updated.

> > This pull request **introduces 2 alerts** when merging [6c824fe](https://github.com/Commonjava/mrrc-uploader/commit/6c824feb8ed0f3c28851a3d7907b02ae6831e5cc) into [52a797b](https://github.com/Commonjava/mrrc-uploader/commit/52a797bb333dc38293a41aae3d5c358ab5508bdc) - [view on LGTM.com](https://lgtm.com/projects/g/Commonjava/mrrc-uploader/rev/pr-698fa31fb8c0e23e11e11d1f4c5542a3dafafbd7)
> > **new alerts:**
> > 
> > * 2 for Module is imported with 'import' and 'import from'
> 
> Hi @yma88 , I've added this LGTM analysis to find any issues in the code automatically. Currently it shows two issues please take a look at them :)

ligangty commented on mrrc/storage/s3client.py:
```diff
@@ -193,6 +193,19 @@ def get_files(self, bucket_name=None, prefix=None, suffix=None)-> List[str]:
             files = [i.key for i in objs]    
         return files
 
+    def download_file(self, bucket_name=None, key=None, download_path=None):
```
Maybe something you need is just read the file content but not download it. You can use these code to do it:
fileObject = bucket.Object(path)
return str(fileObject.get()['Body'].read(), 'utf-8')

ligangty reviewed (commented):
Generally looks good except some comments.

ligangty commented on tests/test_npm_meta_s3.py:
```diff
@@ -0,0 +1,135 @@
+
+    def test_store_package_metadata_to_S3_for_old_version(self):
+        bucket = self.mock_s3.Bucket(MY_BUCKET)
+        original_old_version_package_json = \
```
Suggest to use python string literal here: """ content """. This can avoid character escape (like \\" here) and support line break too, which will make your string content more readable.

yma88 replied:
@ligangty thank you, copy that.

ligangty commented on tests/test_npm_meta_s3.py:
```diff
@@ -0,0 +1,135 @@
+
+    def test_store_package_metadata_to_S3_for_new_version(self):
+        bucket = self.mock_s3.Bucket(MY_BUCKET)
+        original_new_version_package_json = \
```
python string literals.
